tests/pdctl/member/member_test.go

```diff
@@ -116,4 +118,75 @@ func (s *memberTestSuite) TestMember(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(len(members.Members), Equals, 2)
 	c.Succeed()
+
+}
+
+func (s *memberTestSuite) TestUpdateMemberDcLocationInfo(c *C) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	cluster, err := tests.NewTestCluster(ctx, 3, withDcLocation)
+	c.Assert(err, IsNil)
+	err = cluster.RunInitialServers()
+	c.Assert(err, IsNil)
+	cluster.WaitLeader()
+	defer cluster.Destroy()
+
+	pdAddr := cluster.GetConfig().GetClientURL()
+	cmd := pdctl.InitCommand()
+
+	checkDcLocation := func(expected, unexpected []string) {
+		testutil.WaitUntil(c, func(c *C) bool {
+			args := []string{"-u", pdAddr, "member"}
+			_, output, err := pdctl.ExecuteCommandC(cmd, args...)
+			c.Assert(err, IsNil)
+			resp := &pdpb.GetMembersResponse{}
+			c.Assert(json.Unmarshal(output, resp), IsNil)
+			c.Log(fmt.Sprintf("allocator leaders: %s", resp.GetTsoAllocatorLeaders()))
+
+			for _, l := range unexpected {
+				_, ok := resp.GetTsoAllocatorLeaders()[l]
+				if ok {
+					return false
+				}
+			}
+			for _, l := range expected {
+				_, ok := resp.GetTsoAllocatorLeaders()[l]
+				if !ok {
+					return false
+				}
+			}
+			return true
+		}, testutil.WithSleepInterval(time.Second))
+	}
+
+	// check initial dc location
+	checkDcLocation([]string{"pd1", "pd2", "pd3"}, []string{})
+
+	// modify dc location of pd1
+	args := []string{"-u", pdAddr, "member", "dc_location", "name", "pd1", "pd4"}
+	_, output, err := pdctl.ExecuteCommandC(cmd, args...)
+	c.Assert(err, IsNil)
+	c.Assert(strings.Contains(string(output), "Success"), IsTrue)
+	// check dc location after modifictation
+	checkDcLocation([]string{"pd2", "pd3", "pd4"}, []string{"pd1"})
+
+	// tso allocator leader of "pd4" should be pd1 finally.
+	testutil.WaitUntil(c, func(c *C) bool {
+		args := []string{"-u", pdAddr, "member"}
+		_, output, err := pdctl.ExecuteCommandC(cmd, args...)
+		c.Assert(err, IsNil)
+		resp := &pdpb.GetMembersResponse{}
+		c.Assert(json.Unmarshal(output, resp), IsNil)
+
+		m := resp.GetTsoAllocatorLeaders()["pd4"]
+		return m.GetName() == "pd1"
+
+	}, testutil.WithSleepInterval(time.Second))
+}
+
+func withDcLocation(conf *config.Config, serverName string) {
+	conf.LocalTSO = config.LocalTSOConfig{
+		EnableLocalTSO: true,
+		DCLocation:     serverName,
```

Comment:
It's kind of confusing to use `pd*` as `dc-location` here. I think `dc-*` is better.